services/wallet/service.go

```diff
@@ -518,54 +522,54 @@ func (service *Service) LinkGeminiWallet(ctx context.Context, walletID uuid.UUID
+	// If there is no prior linking then we should return the original error.
 	accountID, country, err := geminiClient.ValidateAccount(ctx, verificationToken, depositID)
 	if err != nil {
-		// check if this gemini accountID has already been linked to this wallet,
 		if errors.Is(err, errorutils.ErrInvalidCountry) {
-			ok, priorLinkingErr := service.Datastore.HasPriorLinking(
-				ctx, walletID, uuid.NewV5(ClaimNamespace, accountID))
+			hasPriorLinking, priorLinkingErr := service.Datastore.HasPriorLinking(ctx, walletID, uuid.NewV5(ClaimNamespace, accountID))
 			if priorLinkingErr != nil && !errors.Is(err, sql.ErrNoRows) {
-				return fmt.Errorf("failed to check prior linkings: %w", priorLinkingErr)
+				return "", fmt.Errorf("failed to check prior linkings: %w", priorLinkingErr)
 			}
-			if !ok {
-				// then pass back the original geo error
-				return fmt.Errorf("failed to validate account: %w", err)
+
+			if !hasPriorLinking {
+				return "", fmt.Errorf("failed to validate account: %w", err)
 			}
-			// allow invalid country if there was a prior linking
+
 		} else {
 			// not err invalid country error
-			return fmt.Errorf("failed to validate account: %w", err)
+			return "", fmt.Errorf("failed to validate account: %w", err)
 		}
 	}
 
 	// we assume that since we got linking_info(VerificationToken) signed from Gemini that they are KYC
 	providerLinkingID := uuid.NewV5(ClaimNamespace, accountID)
-	// tx.Destination will be stored as UserDepositDestination in the wallet info upon linking
 	err = service.Datastore.LinkWallet(ctx, walletID.String(), depositID, providerLinkingID, nil, "gemini", country)
 	if err != nil {
-		status := http.StatusInternalServerError
-		if errors.Is(err, ErrTooManyCardsLinked) {
-			status = http.StatusConflict
-		}
 		if errors.Is(err, ErrUnusualActivity) {
-			return handlers.WrapError(err, "unable to link - unusual activity", http.StatusBadRequest)
+			return "", handlers.WrapError(err, "unable to link - unusual activity", http.StatusBadRequest)
 		}
+
 		if errors.Is(err, ErrGeoResetDifferent) {
-			return handlers.WrapError(err, "mismatched provider account regions", http.StatusBadRequest)
+			return "", handlers.WrapError(err, "mismatched provider account regions", http.StatusBadRequest)
 		}
-		return handlers.WrapError(err, "unable to link gemini wallets", status)
+
+		status := http.StatusInternalServerError
+		if errors.Is(err, ErrTooManyCardsLinked) {
+			status = http.StatusConflict
+		}
+
+		return "", handlers.WrapError(err, "unable to link gemini wallets", status)
 	}
-	return nil
+
+	return country, nil
 }
 
 // LinkWallet links a wallet and transfers funds to newly linked wallet
-func (service *Service) LinkWallet(
-	ctx context.Context,
-	wallet uphold.Wallet,
-	transaction string,
-	anonymousAddress *uuid.UUID,
-) error {
+func (service *Service) LinkWallet(ctx context.Context, wallet uphold.Wallet, transaction string,
+	anonymousAddress *uuid.UUID) (string, error) {
```

Comment:
Nit: previous formatting style was how it's usually done in Go for long signatures, this one is more C or C++. Looks a bit foreign. Would be better to keep the previous or just has it as one long line.

Reply:
I think [Go is a little loose](https://google.github.io/styleguide/go/guide#line-length) on the line length and formatting of long lines, personally I find the it very hard to read (in any language) functions which are bunched on top of each other like it was previously and favour the long line approach or as above the more historical 80 chars. 

Changed it to long line approach.